accepted/2022/simplify-rid-model.md

```diff
@@ -0,0 +1,244 @@
+
+| RID        | Description of scope |
+|------------|----------------------|
+| any        | The root RID, which is compatible with all other RIDs. |
+| alpine     | All Alpine versions and architecture builds. |
+| alpine-x64 | All Alpine versions, for x64. Other processor-types are also supported. |
+| alpine3.16-arm64 | Alpine 3.16, for Arm64. Other processor-types are also supported.|
+| centos     | CentOS has a similar scheme as Alpine. |
+| debian     | Debian has a similar scheme as Alpine. |
+| osx        | macOS has a similar scheme as Alpine. |
+| rhel       | Red Hat Enterprise Linux has a similar scheme as Alpine. |
+| tizen      | Tizen has a similar scheme as Alpine. |
+| ubuntu     | Ubuntu has a similar scheme as Alpine. |
+
+Note: Many other Linux distros are represented in `runtimes.json`.
+
+Note: `runtimes.json` will be frozen, which means that RID schemes only supported by this file are abandoned. For example, RIDs that include OS versions will not be updated going forward.
+
+## Source-build
+
+A major design tenet of the RID system is maximizing portability of code. This is particularly true for Linux. That makes sense from the perspective of Microsoft wanting to make one build of .NET available across many Linux distros, separately for both glibc and musl. It makes less sense for distros themselves building .NET from source and publishing binaries to a distro package feed.
+
+We sometimes refer to `linux-x64` (and equally to `linux-arm64` and `linux-x86`) as the "portable Linux RID". As mentioned in the libc section, this RID establishes a wide glibc compatibility range. In addition, the RID also establishes broad compatibility across distros (and their associated versions) for .NET dependencies, like OpenSSL and ICU. The way this particular form of compatibility shows up is observable but is nuanced (and isn't explained here).
+
+We'll use Red Hat as an example of an organization that builds .NET from source, to complete this discussion. They have raised concerns on [source-build not playing nicely with `linux-x64`](https://github.com/dotnet/runtime/pull/62942#issuecomment-1056942235).
+
+For Red Hat, it makes sense to accept and support `linux-x64` assets, but not to produce or offer them. Instead, Red Hat would want to produce `rhel-x64` assets. It's easiest to describe this design-point in terms of concrete scenarios.
+
+**NuGet packages** -- NuGet authors will typically value breadth and efficiency. For example, a package like `SkiaSharp` might target `linux-x64`, but not `rhel-x64`. There is no technical reason for that package to include distro-specific assets. Also, if the author produced a `rhel-x64` asset, they would need to consider producing an `ubuntu-x64` and other similar assets for other distros, and that's not scalable. As a result, it makes sense for Red Hat to support `linux-x64`. There is no technical downside to Red Hat doing so.
```

Comment:
> As a result, it makes sense for Red Hat to support `linux-x64`. There is no technical downside to Red Hat doing so.

Red Hat doesn't do anything to support `linux-x64`. It is people publishing `linux-x64` assets (like Microsoft) that declare compatibility with a range of glibc based Linux distros (like RHEL).

Reply:
I think we're talking past one another.

The question is:

- When a dev using the RHEL variant of .NET restores a NuGet asset with a `linux-x64` as an asset (and no other Linux compatible asset), will they get it?

The `linux-x64` pack assets that Microsoft publishes are very different than the ones that the community publishes in NuGet packages.

Reply:
Maybe use RHEL instead of Red Hat.

Using Red Hat makes it sound like we do the work. We don't.
The `SkiaSharp` NuGet authors are the ones doing the work by providing native assets that (should) work across the `linux-x64` distros, including RHEL.

Reply:
Fixed that.